tip-distributor/src/lib.rs

```diff
@@ -460,104 +470,212 @@ pub fn derive_tip_distribution_account_address(
     )
 }
 
+pub const MAX_RETRIES: usize = 5;
+pub const FAIL_DELAY: Duration = Duration::from_millis(100);
+
+/// Returns unprocessed transactions, along with fail count
+pub async fn sign_and_send_transactions_with_retries_multi_rpc(
+    signer: &Arc<Keypair>,
+    blockhash_rpc_client: &Arc<RpcClient>,
+    rpc_clients: &Arc<Vec<Arc<RpcClient>>>,
+    mut transactions: Vec<Transaction>,
+    max_loop_duration: Duration,
+) -> (Vec<Transaction>, u64) {
+    let error_count = Arc::new(AtomicU64::default());
+    let blockhash = Arc::new(RwLock::new(
+        blockhash_rpc_client
+            .get_latest_blockhash()
+            .await
+            .expect("fetch latest blockhash"),
+    ));
+    let mut rng = rand::thread_rng();
+    transactions.shuffle(&mut rng); // shuffle to avoid racing for the same order of txns as other claim-tip processes
+    let (tx, rx) = async_channel::bounded::<Transaction>(2 * rpc_clients.len());
+    let dispatcher_handle = {
+        let blockhash_rpc_client = blockhash_rpc_client.clone();
+        let blockhash = blockhash.clone();
+        tokio::spawn(async move {
+            let start = Instant::now();
+            let mut last_blockhash_update = Instant::now();
+            while start.elapsed() < max_loop_duration && !transactions.is_empty() {
+                // ensure we always have a recent blockhash
+                if last_blockhash_update.elapsed() > Duration::from_secs(30) {
```

Comment:
i think we should update this more frequently.

blockhashes expire in 150 blocks. finalized typically ~30 slots behind. so 120 slots or ~48s we can expect finalized blockhases to be valid for. assuming we're signing things close to 30s expiry, the tx will only be valid for ~18s

Reply:
can do it like every 2-3s. pretty lightweight rpc call